packages/arb-token-bridge-ui/src/hooks/useCompleteMultiChainTransactions.ts

```diff
@@ -0,0 +1,323 @@
+  // },
+  // Testnet
+  {
+    parentChain: ChainId.Goerli,
+    chain: ChainId.ArbitrumGoerli
+  },
+  {
+    parentChain: ChainId.Sepolia,
+    chain: ChainId.ArbitrumSepolia
+  }
+  // Orbit
+  // {
+  //   parentChain: ChainId.ArbitrumGoerli,
+  //   chain: ChainId.XaiTestnet
+  // }
+  // {
+  //   parentChain: ChainId.ArbitrumSepolia,
+  //   chain: ChainId.StylusTestnet
+  // }
+]
+
+function isWithdrawalFromSubgraph(
+  tx: Withdrawal
+): tx is FetchWithdrawalsFromSubgraphResult & AdditionalTransferProperties {
+  return tx.source === 'subgraph'
+}
+
+function isDeposit(tx: DepositOrWithdrawal): tx is Deposit {
+  return tx.direction === 'deposit'
+}
+
+async function transformTransaction(tx: DepositOrWithdrawal) {
+  const parentChainProvider = getProvider(tx.parentChainId)
+  const chainProvider = getProvider(tx.chainId)
+
+  if (isDeposit(tx)) {
+    return transformDeposit(
+      await updateAdditionalDepositData({
+        depositTx: tx,
+        l1Provider: parentChainProvider,
+        l2Provider: chainProvider
+      })
+    )
+  }
+
+  let withdrawal: L2ToL1EventResultPlus | undefined
+
+  if (isWithdrawalFromSubgraph(tx)) {
+    withdrawal = await mapWithdrawalToL2ToL1EventResult({
+      withdrawal: tx,
+      l1Provider: parentChainProvider,
+      l2Provider: chainProvider
+    })
+  } else {
+    if (isTokenWithdrawal(tx)) {
+      withdrawal = await mapTokenWithdrawalFromEventLogsToL2ToL1EventResult({
+        result: tx,
+        l1Provider: parentChainProvider,
+        l2Provider: chainProvider
+      })
+    } else {
+      withdrawal = await mapETHWithdrawalToL2ToL1EventResult({
+        event: tx,
+        l1Provider: parentChainProvider,
+        l2Provider: chainProvider
+      })
+    }
+  }
+
+  if (withdrawal) {
+    return transformWithdrawal(withdrawal)
+  }
+}
+
+function getProvider(chainId: ChainId) {
+  const rpcUrl =
+    rpcURLs[chainId] ?? getWagmiChain(chainId).rpcUrls.default.http[0]
+  return new StaticJsonRpcProvider(rpcUrl)
+}
+
+/**
+ * Fetches transaction history only, and only for a specific direction.
+ * The query could be e.g. deposits or withdrawals.
+ */
+const useTransactionListByDirection = (
+  direction: 'deposits' | 'withdrawals'
+) => {
+  const [transactions, setTransactions] = useState<DepositOrWithdrawal[][]>([])
+  const [page, setPage] = useState(0)
+  const [loading, setLoading] = useState(true)
+
+  const { address } = useAccount()
+
+  const shouldFetchNextPageForChainPair = useCallback(
+    (chainPairIndex: number) => {
+      if (page === 0) {
+        return true
+      }
+
+      const txCountForChainPair = transactions[chainPairIndex]?.length ?? 0
+      // fetch next page for a chain pair if all fetched pages are full
+      // we check for >= in case some txs were included by initiating a transfer
+      return txCountForChainPair / page >= PAGE_SIZE
+    },
+    [page, transactions]
+  )
+
+  const { data, error } = useSWRImmutable(
```

Comment:
Maybe there's a way to replace this with https://swr.vercel.app/docs/pagination.en-US#infinite-loading